routing/index_graph_starter.cpp

```diff
@@ -76,10 +100,46 @@ void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
   }
 }
 
+void IndexGraphStarter::AddZeroLengthOnewayFeature(Joint::Id from, Joint::Id to)
+{
+  if (from == to)
+    return;
+
+  m_jointsOfAdditionalEdges.insert(from);
+  m_jointsOfAdditionalEdges.insert(to);
+  m_additionalZeroEdges.emplace_back(from, to, m_additionalNextFeatureId++);
+}
+
+void IndexGraphStarter::GetAdditionalEdgesList(Joint::Id jointId, bool isOutgoing,
+                                               vector<JointEdge> & edges)
+{
+  if (m_jointsOfAdditionalEdges.count(jointId) == 0)
+    return;
+
+  for (DirectedEdge const e : m_additionalZeroEdges)
+  {
+    if (isOutgoing)
+    {
+      if (e.GetFrom() == jointId)
+        edges.emplace_back(e.GetTo(), 0 /* weight */);
+    }
+    else
+    {
+      if (e.GetTo() == jointId)
+        edges.emplace_back(e.GetFrom(), 0 /* weight */);
+    }
+  }
+}
+
 void IndexGraphStarter::GetEdgesList(Joint::Id jointId, bool isOutgoing, vector<JointEdge> & edges)
 {
   edges.clear();
 
+  // Note. Additional edges adding here may be ingoing or outgoing edges for start or finish.
+  // Or it may be "arrival edges". That mean |jointId| is not start of finish but
+  // a node which is connected with start of finish by an edge.
+  GetAdditionalEdgesList(jointId, isOutgoing, edges);
```

Comment:
> So, there can be Fake edges and Additional edges?

Let's get rid of additional edges (in the terminology) and use IndexGraphStater::<fake edges/features> and IndexGraph::<fake edges/features>